okhttp/src/jvmMain/kotlin/okhttp3/internal/connection/ExchangeFinder.kt

```diff
@@ -92,57 +92,14 @@ class ExchangeFinder(
   }
 
   /**
-   * Finds a connection and returns it if it is healthy. If it is unhealthy the process is repeated
-   * until a healthy connection is found.
-   */
-  @Throws(IOException::class)
-  private fun findHealthyConnection(
-    connectTimeout: Int,
-    readTimeout: Int,
-    writeTimeout: Int,
-    pingIntervalMillis: Int,
-    connectionRetryEnabled: Boolean,
-    doExtensiveHealthChecks: Boolean
-  ): RealConnection {
-    while (true) {
```

Comment:
Removing this loop FEELS GOOD. The previous behavior was that this would loop at most N + 1 times for N unhealthy pooled/reused connections and 1 new route to attempt. By not returning these until after the health check, we save the loop.